lib/hammer_cli_katello/ping.rb

```diff
@@ -44,7 +44,8 @@ class PingCommand < HammerCLIKatello::Command
     def execute
       d = send_request
       print_data d
-      d['status'] != _("FAIL") ? HammerCLI::EX_OK : 1
+      service_failure = d['services'].values.map { |v| v['status'] }.include?(_("FAIL"))
+      service_failure || d['status'] != _("FAIL") ? 1 : HammerCLI::EX_OK
```

Comment:
It looks like the `d['status'] != _("FAIL")` should be `d['status'] == _("FAIL") ` as that will return 1 if true and 0 if false.

And this is up to you, but this one-liner is a bit hard to parse, it would be nice if it was broken up